Update 1.2.9 - Memory Leaks & Improvements

kavinprabhuc opened the pull request:
- Removed global view initialisation.
- Make use of onViewCreate lifecycle callback.

- Convert PaintFragment.java to Kotlin
- Introduce PaintViewModel.kt to fix leaks issue caused by BrushConfigDialog & EraserConfigDialog
- Removed kotlin std lib as Kotlin 1.4 by default has this included
- Refactor of Sticker and Paint fragments
 
- Updated dependencies and gradle
 
- Added a companion object fun in ImageEditorIntentBuilder.kt for simple intent creation with all features

kavinprabhuc committed:
Removed global view initialisation.
Make use of onViewCreate lifecycle callback.

kavinprabhuc committed:
Convert PaintFragment.java to Kotlin
Introduce PaintViewModel.kt to fix leaks issue caused by BrushConfigDialog & EraserConfigDialog
Removed kotlin std lib as Kotlin 1.4 by default has this included
Refactor of Sticker and Paint fragments

kavinprabhuc committed:
Updated dependencies and gradle

kavinprabhuc committed:
Added a companion object fun in ImageEditorIntentBuilder.kt for simple intent creation with all features

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/ImageEditorIntentBuilder.kt:
```diff
@@ -141,5 +141,37 @@ class ImageEditorIntentBuilder @JvmOverloads constructor(private val context: Co
+                    .withSaturationFeature()
+                    .withBeautyFeature()
+                    .withStickerFeature()
+                    .forcePortrait(true) // Add this to force portrait mode (It's set to false by default)
```
why do we need this?

kavinprabhuc replied:
If force portrait is not set the orientation will be locked for editor activity based on what orientation the device was. If user triggers image editor while the device is in landscape, then. image editor will be in landscape too.

pal-tgiannakopoulos replied:
ok, in this case then let's keep the forcePortrait(true) as you initially did

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/ImageEditorIntentBuilder.kt:
```diff
@@ -141,5 +141,37 @@ class ImageEditorIntentBuilder @JvmOverloads constructor(private val context: Co
+                    .withSaturationFeature()
+                    .withBeautyFeature()
+                    .withStickerFeature()
+                    .forcePortrait(true) // Add this to force portrait mode (It's set to false by default)
```
similar, why?

kavinprabhuc replied:
Same as mentioned above

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/ImageEditorIntentBuilder.kt:
```diff
@@ -141,5 +141,37 @@ class ImageEditorIntentBuilder @JvmOverloads constructor(private val context: Co
         const val EDITOR_TITLE = "editor_title"
         const val SUPPORT_ACTION_BAR_VISIBILITY = "support_action_bar_visibility"
+
+        fun getImageEditorIntent(context: Context, imageLocation: Uri, outputPath: String): Intent {
```
better for these functions to be on our code to be more easily changeable

kavinprabhuc replied:
Done

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/adapter/StickerAdapter.java:
```diff
@@ -68,7 +68,12 @@ public void addStickerImages(String folderPath, int stickerCount) {
             String data = (String) v.getTag();
-            stickerFragment.selectedStickerItem(data);
+            mCallback.onStickerSelected(data);
+//            stickerFragment.selectedStickerItem(data);
```
remove comment line pls

kavinprabhuc replied:
Done

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/BaseEditFragment.java:
```diff
@@ -1,23 +1,27 @@
 
-    protected EditImageActivity ensureEditActivity(){
-        if(activity == null){
+    private void ensureEditActivity() {
```
make this protected and get rid of the field maybe?
it's causing a leak

kavinprabhuc replied:
Done. Also updated all the affected places with not null checks

pal-tgiannakopoulos replied:
As talked in chat, we should hold a local variable and not calling the new method all the time

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/RotateFragment.java:
```diff
@@ -49,27 +53,24 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
-    public void onActivityCreated(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
+    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
```
double imports with jetbrains nullable and non-not-null

kavinprabhuc replied:
Fixed

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/StickerFragment.java:
```diff
@@ -59,66 +57,63 @@ public void onCreate(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
-        this.stickerView = activity.stickerView;
-        flipper = mainView.findViewById(R.id.flipper);
+    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
```
same problem with imports

kavinprabhuc replied:
Fixed

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/StickerFragment.java:
```diff
@@ -59,66 +57,63 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
```
empty implementation

kavinprabhuc replied:
Removed

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/crop/CropFragment.java:
```diff
@@ -128,16 +126,17 @@ private int getColorFromRes(@ColorRes int resId) {
     @Override
-    public void onActivityCreated(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
+    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
```
same problem with imports

kavinprabhuc replied:
Fixed

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/paint/PaintFragment.kt:
```diff
@@ -0,0 +1,187 @@
+                        return@flatMap Single.error<Bitmap>(Throwable("Error occurred while applying paint"))
+                    } else {
+                        return@flatMap Single.just(bitmap)
+                    }
```
this seems to terminate the flatmap early in every case

kavinprabhuc replied:
This seems to be working fine. We had to return the bitmap or the error

kavinprabhuc committed:
Remove intent builder companion object to improve the accessibility of features from the source of use

kavinprabhuc committed:
Remove the global variable which holds the activity instance.
Introduced a protected function to get the activity instance.
Modified all the required fragment and classes

kavinprabhuc committed:
Addressed all the @Nullable and @NotNull duplication issues across fragment classes

kavinprabhuc committed:
Removed empty callback implementation

kavinprabhuc committed:
Removed unused code for orientation changes

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/EditImageActivity.java:
```diff
@@ -450,30 +452,7 @@ protected void onDestroy() {
-            return;
         }
-
-        if (lock) {
```
ok this will not create a problem for us since we will never run this on lower than 18 but I think that it is wrong to completely remove these.
You could change the pattern with an early return though and put an else on the if in line 451 but please keep this code for lower platforms as it will makes sense if we want to do a PR to contribute back to the open-source project

kavinprabhuc committed:
Reverted the screen orientation implementation and added else statement removing the return

kavinprabhuc committed:
Rename .java to .kt

kavinprabhuc committed:
Convert BaseEditFragment to Kotlin and access the activity instance as a local variable

kavinprabhuc committed:
Optimized Imports

pal-tgiannakopoulos commented on ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/RotateFragment.java:
```diff
@@ -119,79 +114,100 @@ public void onClick(View v) {
+                        .doFinally(() -> {
+                            EditImageActivity activityInstance;
+                            if ((activityInstance = getActivityInstance()) != null)
+                                activityInstance.showLoadingDialog();
```
is this change deliberate?
before this change the "doFinally" was running dismiss on the dialog, not show

kavinprabhuc replied:
It should be dismiss, I'll fix and update the same

kavinprabhuc committed:
Fixed dialog loading issue in RotateFragment
